app/models/dataset/WKRemoteSegmentAnythingClient.scala

```diff
@@ -10,23 +10,30 @@ import javax.inject.Inject
 
 class WKRemoteSegmentAnythingClient @Inject()(rpc: RPC, conf: WkConf) {
 
-  def getEmbedding(imageData: Array[Byte],
-                   elementClass: ElementClass.Value,
-                   intensityMin: Option[Float],
-                   intensityMax: Option[Float]): Fox[Array[Byte]] = {
-    val metadataLengthInBytes = 1 + 1 + 4 + 4
-    val buffer = ByteBuffer.allocate(metadataLengthInBytes + imageData.length)
+  def getMask(imageData: Array[Byte],
+              elementClass: ElementClass.Value,
+              selectionTopLeftX: Int,
+              selectionTopLeftY: Int,
+              selectionBottomRightX: Int,
+              selectionBottomRightY: Int,
+              intensityMin: Option[Float],
+              intensityMax: Option[Float]): Fox[Array[Byte]] = {
+    val metadataLengthInBytes = 1 + 1 + 4 + 4 + 4 + 4 + 4 + 4 + 1
+    val buffer = ByteBuffer.allocate(metadataLengthInBytes + imageData.length).order(ByteOrder.LITTLE_ENDIAN)
     buffer.put(ElementClass.encodeAsByte(elementClass))
     buffer.put(if (intensityMin.isDefined && intensityMax.isDefined) 1.toByte else 0.toByte)
-    buffer.order(ByteOrder.LITTLE_ENDIAN).putFloat(intensityMin.getOrElse(0.0f))
-    buffer.order(ByteOrder.LITTLE_ENDIAN).putFloat(intensityMax.getOrElse(0.0f))
+    buffer.putFloat(intensityMin.getOrElse(0.0f))
+    buffer.putFloat(intensityMax.getOrElse(0.0f))
+    buffer.putInt(selectionTopLeftX)
+    buffer.putInt(selectionTopLeftY)
+    buffer.putInt(selectionBottomRightX)
+    buffer.putInt(selectionBottomRightY)
+    buffer.put(1.toByte) // section count always 1 for now
```

Comment:
"section count" is some SAM specific parameter? The number of segments / regions to annotate?

Reply:
Yes. In the next iteration, we want to send multiple sections to torch serve and then stream the masks for each section back to the frontend. https://www.notion.so/scalableminds/Design-Doc-Integrate-SAM2-with-Webknossos-7dc37ad919d04f3fb0dc22a1baac7fa0?pvs=4#690e62caae60462091dd67e5c21862b4

Reply:
“section” is what we previously called “slice” (so a subset of the image data with depth=1). We are in the process to adapting to the somewhat-standard word “section” for that.